tests/core/pyspec/eth2spec/test/altair/transition/test_transition.py

```diff
@@ -418,3 +418,37 @@ def test_transition_with_no_attestations_until_after_fork(state, fork_epoch, spe
 
     yield "blocks", blocks
     yield "post", state
+
+
+@with_fork_metas([ForkMeta(pre_fork_name=pre, post_fork_name=post, fork_epoch=2) for pre, post in ALL_PRE_POST_FORKS])
+def test_non_empty_historical_roots(state, fork_epoch, spec, post_spec, pre_tag, post_tag):
+    """
+    Test with non-empty pre-state `state.historical_roots`.
+
+    Since Capella froze `historical_roots`, Capealla spec doesn't invoke `process_historical_roots_update` anymore.
+    Therefore, we need to fill in `historical_roots` with non-empty value.
+    """
+    # fill in historical_roots with non-empty values
+    pre_historical_roots = [b'\x56' * 32 for _ in range(spec.SLOTS_PER_HISTORICAL_ROOT)]
```

Comment:
`SLOTS_PER_HISTORICAL_ROOT` is fine here but it's a bit misleading in it's relationship to `historical_roots` which  is a *list* of unbound length

Reply:
The test would do its job with > 0 items so why no set it to 1?